IssueActivity onBackPressed missing super call

From pull request action build output

IssueActivity.java:273: Error: Overriding method should call super.onBackPressed [MissingSuperCall]
      public void onBackPressed() {
                  ~~~~~~~~~~~~~
  
     Explanation for issues of type "MissingSuperCall":
     Some methods, such as View#onDetachedFromWindow, require that you also call
     the super implementation as part of your method.

--- a/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/AboutActivity.java
+++ b/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/AboutActivity.java
@@ -197,7 +197,7 @@ public class AboutActivity extends AppCompatActivity {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case android.R.id.home:
-                onBackPressed();
+                finish();
                 return true;
         }
         return super.onOptionsItemSelected(item);

--- a/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/IssueActivity.java
+++ b/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/IssueActivity.java
@@ -4,6 +4,8 @@ import android.app.Activity;
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Bundle;
+
+import androidx.activity.OnBackPressedCallback;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
@@ -42,7 +44,6 @@ import org.a5calls.android.a5calls.model.Issue;
 import org.a5calls.android.a5calls.util.AnalyticsManager;
 import org.a5calls.android.a5calls.util.MarkdownUtil;
 
-import java.util.List;
 import java.util.Locale;
 
 /**
@@ -176,6 +177,13 @@ public class IssueActivity extends AppCompatActivity {
         });
 
         new AnalyticsManager().trackPageview(String.format("/issue/%s/", mIssue.slug), this);
+
+        getOnBackPressedDispatcher().addCallback(this, new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                returnToMain();
+            }
+        });
     }
 
     @Override
@@ -260,7 +268,7 @@ public class IssueActivity extends AppCompatActivity {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case android.R.id.home:
-                onBackPressed();
+                returnToMain();
                 return true;
             case R.id.menu_share:
                 sendShare();
@@ -269,10 +277,6 @@ public class IssueActivity extends AppCompatActivity {
         return super.onOptionsItemSelected(item);
     }
 
-    @Override
-    public void onBackPressed() {
-        returnToMain();
-    }
 
     private void returnToMain() {
         Intent intent = new Intent();

--- a/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/StatsActivity.java
+++ b/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/StatsActivity.java
@@ -296,7 +296,7 @@ public class StatsActivity extends AppCompatActivity {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case android.R.id.home:
-                onBackPressed();
+                finish();
                 return true;
             case R.id.menu_share:
                 sendShare();

--- a/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/TutorialActivity.java
+++ b/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/TutorialActivity.java
@@ -51,7 +51,7 @@ public class TutorialActivity extends AppCompatActivity {
                     finish();
                 } else {
                     // Otherwise, select the previous step.
-                    binding.viewPager.setCurrentItem(binding.viewPager.getCurrentItem() - 1);
+                    onPreviousPagePressed();
                 }
             }
         });

--- a/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/TutorialActivity.java
+++ b/5calls/app/src/main/java/org/a5calls/android/a5calls/controller/TutorialActivity.java
@@ -2,6 +2,8 @@ package org.a5calls.android.a5calls.controller;
 
 import android.content.Intent;
 import android.os.Bundle;
+
+import androidx.activity.OnBackPressedCallback;
 import androidx.annotation.Nullable;
 import com.google.android.material.snackbar.Snackbar;
 import androidx.fragment.app.Fragment;
@@ -40,22 +42,30 @@ public class TutorialActivity extends AppCompatActivity {
 
         binding.viewPager.setAdapter(new TutorialPagerAdapter(getSupportFragmentManager()));
 
+        getOnBackPressedDispatcher().addCallback(this, new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                if (binding.viewPager.getCurrentItem() == 0) {
+                    // If the user is currently looking at the first step, allow the system to handle the
+                    // Back button. Call finish() on this activity and pop the back stack.
+                    finish();
+                } else {
+                    // Otherwise, select the previous step.
+                    binding.viewPager.setCurrentItem(binding.viewPager.getCurrentItem() - 1);
+                }
+            }
+        });
+
         new AnalyticsManager().trackPageview("/tutorial", this);
     }
 
-    @Override
-    public void onBackPressed() {
-        if (binding.viewPager.getCurrentItem() == 0) {
-            // If the user is currently looking at the first step, allow the system to handle the
-            // Back button. This calls finish() on this activity and pops the back stack.
-            super.onBackPressed();
-        } else {
-            // Otherwise, select the previous step.
+    public void onPreviousPagePressed() {
+        if (binding.viewPager.getCurrentItem() > 0) {
             binding.viewPager.setCurrentItem(binding.viewPager.getCurrentItem() - 1);
         }
     }
 
-    public void onNextPressed() {
+    public void onNextPagePressed() {
         binding.viewPager.setCurrentItem(binding.viewPager.getCurrentItem() + 1);
     }
 
@@ -97,7 +107,7 @@ public class TutorialActivity extends AppCompatActivity {
             rootView.findViewById(R.id.btn_next).setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    ((TutorialActivity) getActivity()).onNextPressed();
+                    ((TutorialActivity) getActivity()).onNextPagePressed();
                 }
             });
             return rootView;
@@ -119,13 +129,13 @@ public class TutorialActivity extends AppCompatActivity {
             rootView.findViewById(R.id.btn_back).setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    getActivity().onBackPressed();
+                    ((TutorialActivity) getActivity()).onPreviousPagePressed();
                 }
             });
             rootView.findViewById(R.id.btn_next).setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    ((TutorialActivity) getActivity()).onNextPressed();
+                    ((TutorialActivity) getActivity()).onNextPagePressed();
                 }
             });
             return rootView;
@@ -209,7 +219,7 @@ public class TutorialActivity extends AppCompatActivity {
             rootView.findViewById(R.id.btn_back).setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    getActivity().onBackPressed();
+                    ((TutorialActivity) getActivity()).onPreviousPagePressed();
                 }
             });
             rootView.findViewById(R.id.get_started_btn).setOnClickListener(

